perfkitbenchmarker/providers/digitalocean/digitalocean_virtual_machine.py

```diff
@@ -98,132 +74,86 @@ def _Create(self):
-    destroy = [v for v in response if v['type'] == 'destroy'][0]
 
-    # Wait for completion. Actions are global objects, so the action
-    # status should still be retrievable after the VM got destroyed.
-    # We don't care about the result, let the _Exists check decide if we
-    # need to try again.
-    self._GetActionResult(destroy['id'])
+    response, retcode = util.DoctlAndParse(
+        ['compute', 'droplet', 'delete', self.droplet_id])
+    # The command doesn't return the HTTP status code, and the error
+    # format is very difficult to parse, so we string search.
+    if retcode and '404' in response['errors'][0]['detail']:
+      return
+    elif retcode:
+      raise errors.Resource.RetryableDeletionError('Deletion failed: %s' %
+                                                   (response,))
 
   def _Exists(self):
     """Returns true if the VM exists."""
-    _, ret = util.RunCurlCommand(
-        'GET', 'droplets/%s' % self.droplet_id,
-        suppress_warning=True)
-    if ret == 0:
-      return True
-    if ret == 404:
-      # Definitely doesn't exist.
-      return False
-    # Unknown status - assume it doesn't exist. TODO(klausw): retry?
-    return False
+
+    response, retcode = util.DoctlAndParse(
+        ['compute', 'droplet', 'get', self.droplet_id])
+
+    return retcode == 0
 
   def CreateScratchDisk(self, disk_spec):
     """Create a VM's scratch disk.
 
     Args:
       disk_spec: virtual_machine.BaseDiskSpec object of the disk.
     """
-    if disk_spec.disk_type != disk.STANDARD:
-      # TODO(klausw): support type BOOT_DISK once that's implemented.
-      raise errors.Error('DigitalOcean does not support disk type %s.' %
-                         disk_spec.disk_type)
-
-    if self.scratch_disks:
-      # We have a "disk" already, don't add more. TODO(klausw): permit
-      # multiple creation for type BOOT_DISK once that's implemented.
-      raise errors.Error('DigitalOcean does not support multiple disks.')
-
-    # Just create a local directory at the specified path, don't mount
-    # anything.
-    self.RemoteCommand('sudo mkdir -p {0} && sudo chown -R $USER:$USER {0}'
-                       .format(disk_spec.mount_point))
-    self.scratch_disks.append(digitalocean_disk.DigitalOceanDisk(disk_spec))
+
+    if disk_spec.disk_type == disk.LOCAL:
+      if self.scratch_disks and self.scratch_disks[0].disk_type == disk.LOCAL:
+        # We have a disk object already, don't add more.
+        raise errors.Error('DigitalOcean does not support multiple disks.')
+
+      if disk_spec.num_striped_disks != 1:
+        raise ValueError('num_striped_disks=%s, but DigitalOcean VMs can only '
+                         'have one local disk.' % disk_spec.num_striped_disks)
+      # Since the local disk is already mounted, just create a local
+      # directory at the specified path. Calling
+      # self._CreateScratchDiskFromDisks would try to mount the
+      # volume, so don't do that.
```

Comment:
Add a comment explain why we want to mock boot disk as data disk (run on local ssds)